tests/php/test-class-amp-theme-support.php

```diff
@@ -1203,9 +1206,56 @@ function() {
 		$body_classes = get_body_class();
 		$this->assertContains( 'customize-support', $body_classes );
 		$this->assertNotContains( 'no-customize-support', $body_classes );
+	}
 
-		$this->assertEquals( 10, has_filter( 'style_loader_tag', [ 'AMP_Theme_Support', 'filter_admin_bar_style_loader_tag' ] ) );
-		$this->assertEquals( 10, has_filter( 'script_loader_tag', [ 'AMP_Theme_Support', 'filter_admin_bar_script_loader_tag' ] ) );
+	/**
+	 * Test init_admin_bar to ensure dashicons are not added to dev mode when directly enqueued.
+	 *
+	 * @covers \AMP_Theme_Support::init_admin_bar()
+	 * @covers \AMP_Theme_Support::filter_admin_bar_style_loader_tag()
+	 */
+	public function test_init_admin_bar_for_directly_enqueued_dashicons() {
+		require_once ABSPATH . WPINC . '/class-wp-admin-bar.php';
+
+		global $wp_admin_bar;
+		$wp_admin_bar = new WP_Admin_Bar();
+		$wp_admin_bar->initialize();
+		AMP_Theme_Support::init_admin_bar();
+
+		// Enqueued directly.
+		wp_enqueue_style( 'dashicons' );
+
+		ob_start();
+		wp_print_styles();
+		$output = ob_get_clean();
+
+		$this->assertContains( '<link rel=\'stylesheet\' id=\'dashicons-css\'', $output ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedStylesheet
+		$this->assertContains( '<link data-ampdevmode rel=\'stylesheet\' id=\'admin-bar-css\'', $output ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedStylesheet
+	}
+
+	/**
+	 * Test init_admin_bar to ensure dashicons are not added to dev mode when indirectly enqueued.
+	 *
+	 * @covers \AMP_Theme_Support::init_admin_bar()
+	 * @covers \AMP_Theme_Support::filter_admin_bar_style_loader_tag()
+	 */
+	public function test_init_admin_bar_for_indirectly_enqueued_dashicons() {
+		require_once ABSPATH . WPINC . '/class-wp-admin-bar.php';
+
+		global $wp_admin_bar;
+		$wp_admin_bar = new WP_Admin_Bar();
+		$wp_admin_bar->initialize();
+		AMP_Theme_Support::init_admin_bar();
+
+		// Enqueued indirectly.
+		wp_enqueue_style( 'my-font-pack', 'https://example.com/fonts', [ 'dashicons' ], '0.1' );
+
+		ob_start();
+		wp_print_styles();
+		$output = ob_get_clean();
+
+		$this->assertContains( '<link rel=\'stylesheet\' id=\'dashicons-css\'', $output ); // phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedStylesheet
```

Comment:
Maybe add another assert here to make sure you don't have both the dev-mode and the non-dev-mode dashicons enqueue here?

Reply:
Fixed in https://github.com/ampproject/amp-wp/pull/3187/commits/290adf71350fb758be61c03bef9c94c208ee6761